### PR TITLE
#1125 feat: migration + model + test

### PR DESCRIPTION
**Pull request by stefanorosanelli**
This PR solves #1125

### CODE REVIEW

File: plugins/BEdita/Core/config/Migrations/20170307123150_CreateDateRanges.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                'null' => true,
+            ])
+            ->addIndex(
+                [
+                    'object_id',
+                ],
+                [
+                    'name' => 'dateranges_objectid_idx',
+                ]
+            )
```
Comment (fquffio) [inline plugins/BEdita/Core/config/Migrations/20170307123150_CreateDateRanges.php:52]: I would add indexes on `start_date` and `end_date` columns as well.

File: plugins/BEdita/Core/tests/TestCase/Model/Table/DateRangesTableTest.php [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    /**
+     * Fixtures
+     *
+     * @var array
+     */
+    public $fixtures = [
+        'plugin.BEdita/Core.objects',
+        'plugin.BEdita/Core.object_types',
+        'plugin.BEdita/Core.date_ranges',
+        // 'plugin.b_edita/core.properties',
```
Comment (batopa) [inline plugins/BEdita/Core/tests/TestCase/Model/Table/DateRangesTableTest.php]: I would remove the comments

File: plugins/BEdita/Core/src/Model/Entity/DateRange.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<?php
+namespace BEdita\Core\Model\Entity;
```
Comment (batopa) [inline plugins/BEdita/Core/src/Model/Entity/DateRange.php]: Missing BEdita disclaimer

File: plugins/BEdita/Core/config/Migrations/20170307123150_CreateDateRanges.php (hunk 1 above)
Comment (stefanorosanelli) [reply plugins/BEdita/Core/config/Migrations/20170307123150_CreateDateRanges.php:52]: done

File: plugins/BEdita/Core/src/Model/Table/DateRangesTable.php [hunk 4]
```diff
@@ -0,0 +1,88 @@
+<?php
+namespace BEdita\Core\Model\Table;
```
Comment (batopa) [inline plugins/BEdita/Core/src/Model/Table/DateRangesTable.php]: Missing BEdita disclaimer

File: plugins/BEdita/Core/tests/Fixture/DateRangesFixture.php [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<?php
+namespace BEdita\Core\Test\Fixture;
```
Comment (batopa) [inline plugins/BEdita/Core/tests/Fixture/DateRangesFixture.php]: Missing BEdita disclaimer

File: plugins/BEdita/Core/tests/TestCase/Model/Table/DateRangesTableTest.php [hunk 6]
```diff
@@ -0,0 +1,108 @@
+<?php
+namespace BEdita\Core\Test\TestCase\Model\Table;
```
Comment (batopa) [inline plugins/BEdita/Core/tests/TestCase/Model/Table/DateRangesTableTest.php]: Missing BEdita disclaimer

File: plugins/BEdita/Core/src/Model/Entity/DateRange.php (hunk 3 above)
Comment (stefanorosanelli) [reply plugins/BEdita/Core/src/Model/Entity/DateRange.php]: added

File: plugins/BEdita/Core/src/Model/Table/DateRangesTable.php (hunk 4 above)
Comment (stefanorosanelli) [reply plugins/BEdita/Core/src/Model/Table/DateRangesTable.php]: added

File: plugins/BEdita/Core/tests/Fixture/DateRangesFixture.php (hunk 5 above)
Comment (stefanorosanelli) [reply plugins/BEdita/Core/tests/Fixture/DateRangesFixture.php]: done

File: plugins/BEdita/Core/tests/TestCase/Model/Table/DateRangesTableTest.php (hunk 6 above)
Comment (stefanorosanelli) [reply plugins/BEdita/Core/tests/TestCase/Model/Table/DateRangesTableTest.php]: done

File: plugins/BEdita/Core/tests/TestCase/Model/Table/DateRangesTableTest.php (hunk 2 above)
Comment (stefanorosanelli) [reply plugins/BEdita/Core/tests/TestCase/Model/Table/DateRangesTableTest.php]: done

**Pull request merged**